src/lib/mongoBackend/mongoListSubscriptions.cpp

```diff
@@ -23,42 +23,171 @@
 * Author: Orion dev team
 */
 
+#include "common/sem.h"
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
 #include "mongoBackend/mongoListSubscriptions.h"
+#include "mongoBackend/MongoGlobal.h"
+#include "mongoBackend/collectionOperations.h"
+
+#include "mongo/client/dbclient.h"
+
+using namespace ngsiv2;
+
+/* ****************************************************************************
+*
+* setSubscriptionId -
+*/
+extern void setSubscriptionId(Subscription* s, const BSONObj& r)
+{
+  s->id = r.getField("_id").OID().toString();
+}
+
+
+/* ****************************************************************************
+*
+* setSubject -
+*/
+extern void setSubject(Subscription* s, const BSONObj& r)
+{
+  // Entities
+  std::vector<BSONElement> ents = r.getField(CSUB_ENTITIES).Array();
+  for (unsigned int ix = 0; ix < ents.size(); ++ix)
+  {
+    BSONObj ent           = ents[ix].embeddedObject();
+    std::string id        = ent.getStringField(CSUB_ENTITY_ID);
+    std::string type      = ent.getStringField(CSUB_ENTITY_TYPE);
+    std::string isPattern = ent.getStringField(CSUB_ENTITY_ISPATTERN);
+
+    EntID en;
+    if (isFalse(isPattern))
+    {
+      en.id = id;
+    }
+    else
+    {
+      en.idPattern = id;
+    }
+    en.type = type;
+
+    s->subject.entities.push_back(en);
+  }
+
+  // Condition
+  std::vector<BSONElement> conds = r.getField(CSUB_CONDITIONS).Array();
+  for (unsigned int ix = 0; ix < conds.size(); ++ix)
+  {
+    BSONObj cond = conds[ix].embeddedObject();
+    // The ONCHANGE check is needed, as a subscription could mix different conditions types in DB
+    if (std::string(cond.getStringField(CSUB_CONDITIONS_TYPE)) == "ONCHANGE")
+    {
+      std::vector<BSONElement> condValues = cond.getField(CSUB_CONDITIONS_VALUE).Array();
+      for (unsigned int jx = 0; jx < condValues.size(); ++jx)
+      {
+        std::string attr = condValues[jx].String();
+        s->subject.condition.attributes.push_back(attr);
+      }
+    }
+  }
+
+  // Note that current DB model is based in NGSIv1 and doesn't consider expressions. Thus
+  // subject.condition.expression cannot be filled. The implemetion will be enhanced once
+  // the DB model gets defined
+  // TBD
```

Comment:
Mmmm... not sure if that would be a good idea. Right now the comment is in the place were the new code to fill the expression field would be in the future.